charts/kubernetes-external-secrets/templates/crd.yaml

```diff
@@ -0,0 +1,118 @@
+{{- if .Values.customResourceManageEnabled | not }}
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: externalsecrets.kubernetes-client.io
+  annotations:
+    app.kubernetes.io/managed-by: helm
```

Comment:
Not sure how to address the two different copies of the CRD... I'm also leveraging this annotation in the E2E tests which is a bit hacky.

Perhaps in 4.x the custom resource manager can be removed in favor of the chart CRD install?

Reply:
So for helm v3 we should probably put the CRD in crds folder, these cant be templated though IIRC.
We could probably source this file in the code as well to avoid copies, and overrwrite `metadata.annotations["app.kubernetes.io/managed-by"]` in the code before upserting it.

For helm v2 this file could be changed to something like

```yaml
{{- if .Values.customResourceManagerDisabled }}
{{- if .Values.crds.create }}
{{- range $path, $bytes := .Files.Glob "crds/*.yaml" }}
{{ $.Files.Get $path }}
---
{{- end }}
{{- end }}
{{- end }}
```

Probably we should also set `DISABLE_CUSTOM_RESOURCE_MANAGER` to `true` by default if running helm v3.

Maybe this is all overkill :D either way we need to support both helm v2 and v3, I'll try to check with silas if he has any opinion on removing the CRD management from the runtime. I think it makes sense to favor the chart installing the CRD.

So perhaps we can go straight for 4.x solution :)

Reply:
Great suggestions for sourcing the CRD and v2 backwards compat, I'll give them a shot.

Reply:
Alright I got the CRD sourced from one location in the repo now, and added your template for helm v2 support. However, I couldn't conditionally set a value depending on whether or not helm v2/v3 was being used so `DISABLE_CUSTOM_RESOURCE_MANAGER` will have to default to `false` for now to remain backwards compat.

The helm docs will need to be updated as a result to include `--skip-crds` and explain how to install the CRD via the chart using v2 and v3.